src/storage/repositories/zip.py

```diff
@@ -24,7 +24,10 @@ def update(self, entity: dict, storage_recipe=None, **kwargs):
         entity.pop("_id", None)
         entity.pop("uid", None)
         entity["__path__"] = entity["__path__"].rstrip("/")
-        write_to = f"{entity['__path__']}/{entity['name']}.json"
+        if "name" not in entity:
+            write_to = f"{entity['__path__']}/unnamed_document.json"
```

Comment:
should this add some random id to allow for multiple unnamed_documents?

Reply:
ye!